ksqldb-engine/src/main/java/io/confluent/ksql/engine/generic/GenericRecordFactory.java

```diff
@@ -75,23 +75,29 @@ public KsqlGenericRecord build(
               + ". Got " + expressions);
     }
 
-    final LogicalSchema schemaWithRowTime = withRowTime(schema);
+    final LogicalSchema schemaWithPseudoColumns = withPseudoColumns(schema, config);
     for (ColumnName col : columns) {
-      if (!schemaWithRowTime.findColumn(col).isPresent()) {
+
+      if (!schemaWithPseudoColumns.findColumn(col).isPresent()) {
         throw new KsqlException("Column name " + col + " does not exist.");
       }
+
+      if (SystemColumns.isSystemColumn(col, config)
```

Comment:
Don't we need to check that the column is a pseudo column, not just a system column? I think confusion like this is why we should simply have `isDisallowedForInsertValues()` accept any column name rather than throw on non-pseudo-columns.

Reply:
I agree, I'll make the method accept any column name.

Reply:
We don't need this anymore, in light of the update you made to `isDisallowedForInsertValues()`.